internal/exec/engine.go

```diff
@@ -162,11 +163,30 @@ func logStructuredJournalEntry(config string, src string, isReferenced bool) err
 	return nil
 }
 
-// acquireConfig returns the configuration, first checking a local cache
-// before attempting to fetch it from the provider.
-func (e *Engine) acquireConfig() (cfg types.Config, err error) {
+// acquireConfig returns the configuration from a local cache. If one
+// does not exist then it will attempt to fetch the configuration from
+// the provider when in fetch stages.
+func (e *Engine) acquireConfig(stageName string) (cfg types.Config, err error) {
+	cfg, err = e.acquireCachedConfig()
+	if err != nil {
+		if os.IsNotExist(err) && strings.HasPrefix(stageName, "fetch") {
```

Comment:
I think for `fetch`/`fetch-offline`, we shouldn't try to even use the cache. Not sure whether we want to take that further and also error out if it's present. But it feels misleading to not actually fetch?

Reply:
That's fair, I've updated it to only attempt to fetch from the provider config in `fetch*` stages.